### PR TITLE
Fix current user issue and add tracking features

### PR DESCRIPTION
**Pull request by ryan-boder**
- Fix bug with outdated use of api._currentUser
- Add ability to exclude groups from tracking
- Add ability to track user ID for cross device tracking
- Add ability to track across subdomains

### CODE REVIEW

File: settings.yml [hunk 1]
```diff
@@ -5,3 +5,15 @@ host_url:
 website_id:
   type: string
   default: ''
+exclude_groups:
+  type: string
+  default: ''
+  description: Comma separated list of groups to exclude from tracking such as admins, staff.
```
Comment (eviltrout) [inline settings.yml]: This is odd - typically we put the descriptions in `server.en.yml` - I see the `host_url` already did it this way, but I imagine it doesn't work in the app. Did you test in a browser to see if the description showed up in the admin section?

Comment (ryan-boder) [reply settings.yml]: I'm not familiar with Discourse code conventions but yes, it works. The descriptions show up in the theme components settings just like host_url. If there's a preferred place to put them I might need you to show me how. There isn't a server.en.yml in this component.

Comment (eviltrout) [reply settings.yml]: It looks like it's my lack of knowledge with the theme component settings then! Apologies.

**Comment by eviltrout** [comment]
Thanks!

**Pull request merged**